### ISSUE TITLE
MultiInputObj NOTHING

### ISSUE THREAD

**Issue by chasejohnson3** [nipype/pydra#405]
I am using Pydra 0.13 and am trying to create commandline parameters using a ShellCommandTask. One of the parameters is of type MultiInputObj. If I do not specify the input for this parameter, I would expect the parameter to not be included in the command call (this is the case if the parameter is of type pydra.specs.File). Instead, the parameter is given with the value "NOTHING". The behavior can be replicated with the following script:

```
from pydra import ShellCommandTask
from pydra.engine.specs import SpecInfo, MultiInputObj, ShellSpec, File
import attr

my_input_spec = SpecInfo(
    name="Input",
    fields=[
        (
            "arg1",
            attr.ib(
                type=MultiInputObj,
                metadata={
                    "argstr": "--arg1",
                    "help_string": "Command line argument 1",
                }
            ),
        ),
        (
            "arg2",
            attr.ib(
                type=MultiInputObj,
                metadata={
                    "argstr": "--arg2",
                    "help_string": "Command line argument 2",
                }
            ),
        ),
        (
            "arg3",
            attr.ib(
                type=File,
                metadata={
                    "argstr": "--arg3",
                    "help_string": "Command line argument 3",
                }
            ),
        ),
    ],
    bases=(ShellSpec,),
)

shelly = ShellCommandTask(
    name="shelly", executable="echo", input_spec=my_input_spec
)

shelly.inputs.arg2 = "argument2"

print(shelly.cmdline)
# Expect   shelly.cmdline == "echo --arg2 argument"
# instead, shelly.cmdline == "echo --arg1 NOTHING --arg2 argument2"
```

As the comments at the bottom describe, I would expect shelly.cmdline to be `echo --arg2 argument` when arg2 is the only input given. However, shelly.cmdline is actually `echo --arg1 NOTHING --arg2 argument2`

This behavior shows up for both MultiInputObj and MultiInputFile types. Is this the intended behavior, or if a MultiInput parameter is not given an input, should it be excluded from the cmdline?

**Comment by djarecka** [comment]
Thanks for reporting! You're right, `arg1` should not be part of the command

**Comment by chasejohnson3** [comment]
Closing because this issue was addressed in PR #406
